packages/shared/src/hooks/useActions.ts

```diff
@@ -45,13 +45,20 @@ export const useActions = (): UseActions => {
     },
   });
 
-  return useMemo<UseActions>(
-    () => ({
+  return useMemo<UseActions>(() => {
+    const checkHasCompleted = (type: ActionType) =>
+      actions?.some((action) => action.type === type && !!action.completedAt);
```

Comment:
This can be outside the return no? Just seems more clean (and then set dependency)